### ISSUE TITLE
Revise the project proposal and add implementation plan

### ISSUE THREAD

**Issue by pseudoyu** [pseudoyu/InvoFlow#1]

**Comment by Tanya-ruby** [comment]
**Workflow of  InvoFlow(Without Bulk Creation Initially)**
**1)User (Seller) Connects Wallet:**
The seller connects their Ethereum wallet (e.g., MetaMask) to the web application configured to support both Sepolia and Polygon networks.

**2)Create Invoice:**

The seller navigates to the "Create Invoice" page.
The seller fills in the buyer's address, the amount to be paid, the due date, and selects the chain (Sepolia or Polygon).
The seller submits the form, which calls the createInvoice function in the InvoiceFactory smart contract.
The contract records the invoice details, including the selected chain, and emits an InvoiceCreated event.

**3)View Invoices:**

The seller navigates to the "My Invoices" page.
The frontend fetches the seller's invoices from the InvoiceFactory contract and displays them.

**4)Pay Invoice:**

The buyer navigates to the "Pay Invoice" page.
The buyer sees the invoice issued to them and the chain on which it was created.
The buyer clicks on "Pay," which calls the payInvoice function in the InvoiceFactory contract.
The contract transfers the specified amount of the selected chain's native currency (MATIC for Polygon or ETH for Sepolia) from the buyer to the seller and marks the invoice as paid.
The contract emits an InvoicePaid event.

**Implementation Plan:**

- Setting Up Foundry Project

- Set Up OpenZeppelin

Creating the Smart Contract

- Create the InvoiceFactory Contract:
Implement functions for creating single invoices.
Implement payment functions that ensure payments are made on the correct chain.

- Writing Tests for the Contract
Use Foundry to write comprehensive tests for the smart contracts.

- Deploying the Contract
Write Deployment Scripts:
Use Foundry to write scripts for deploying the InvoiceFactory contract to Sepolia and Polygon testnets.
Configure the deployment scripts to set initial parameters if needed.

- Deploy Contracts:
Deploy the InvoiceFactory contract to Sepolia and Polygon testnets.

-Building the Frontend Using Remix
Set Up Remix Project
Install Dependencies
Define Routes in Remix

- Creation of Different Pages

1. Wallet Connection:Develop a wallet connection feature using MetaMask.
2. Create Invoice Page
3. View Invoices Page
4. Pay Invoice Page

**Phase 2: Adding Bulk Creation**(later,once the core functionality is done)
@pseudoyu

**Comment by pseudoyu** [comment]
only support native token or can also support some whitelisted ERC20 tokens?

**Comment by Tanya-ruby** [comment]
> only support native token or can also support some whitelisted ERC20 tokens?

We can use both the native tokens and a few select erc20 tokens right?Like USDT
Also what do you think is better,We will have to support different chains and tokens in those chains.
Another question is do we support cross chain payments for now?Like settlement of invoices by swap and conversion method?We can add this later in the project,since it might increase the complexity in the beginning.
What do you say @pseudoyu

**Comment by pseudoyu** [comment]
> > only support native token or can also support some whitelisted ERC20 tokens?
> 
> We can use both the native tokens and a few select erc20 tokens right?Like USDT Also what do you think is better,We will have to support different chains and tokens in those chains. Another question is do we support cross chain payments for now?Like settlement of invoices by swap and conversion method?We can add this later in the project,since it might increase the complexity in the beginning. What do you say @pseudoyu

yes we add add a extra admin role function in the contract, to add whitelist to approve ERC20 tokens, so we can add universal support for USDT for now and more tokens in the future.

i think it's no need for us to do the cross chain payments for now, users should do conversions by themselves, much cheaper and easier for us to handle.
